docs/conceptual-guides/modular-arch-packages.md

```diff
@@ -0,0 +1,93 @@
+---
+myst:
+  html_meta:
+    "description": "Plone's frontend modular architecture packages."
+    "property=og:description": "Plone's frontend modular architecture packages."
+    "property=og:title": "Plone's frontend modular architecture packages"
+    "keywords": "Plone, modular architecture, packages, frontend, user interface"
+---
+
+# Plone's frontend modular architecture packages
+
+This document gives you a quick overview of the Plone frontend’s modular architecture.
+It explains how the different packages are organized and what role they play in the system.
+
+These packages are expected to be used and become part of Plone 7.
+
+We group these packages into a few categories to make it easier to understand their purpose:
+
+- **Core packages** — the foundation pieces everyone builds on.
+- **Utilities packages** — helpful tools and helpers used by add-on packages.
+- **Add-on packages** — feature packages that add UI elements and functionality.
+- **Development utility packages** — tools to help with building and developing.
+
+
+## `@plone/types`
+
+`@plone/types` is a special package used during development.
+It contains TypeScript type definitions for Plone.
+It's considered a core package, and it's the only package that the other core packages can rely on as a `devDependency` in your project configuration.
+
+This package contains `.d.ts` typing definitions, curated by hand.
+It doesn’t need to be built or bundled.
+It’s published as-is so you can import the type info wherever you need it.
+
+
+## Core packages (level 1)
+
+These are the essential building blocks of the frontend.
+They provide the main features and APIs that other packages rely on.
+They don’t depend on any other `@plone/*` packages except `@plone/types`.
+
+Core packages are published as traditional bundles (transpiled).
+Their bundles work in both CommonJS and ECMAScript Module (ESM) environments.
+
+The core packages are:
+- `@plone/client`
+- `@plone/components`
+- `@plone/registry`
+
+
+## Utilities packages (level 2)
+
+These packages provide useful tools and helpers.
+They can depend on core packages and other utility packages.
+They are also published as traditional bundles that work in CommonJS and ESM environments.
+
+The utility packages are:
+- `@plone/providers`
+- `@plone/helpers`
+- `@plone/react-router`
+
```

Comment:
I'm not sure I really understand the distinction you're trying to make between core and utilities.